cunumeric/_ufunc/ufunc.py

```diff
@@ -571,19 +571,20 @@ def _find_common_type(
         )
         use_min_scalar = not (all_scalars or all_arrays or lossy_conversion)
 
-        scalar_types = []
-        array_types = []
+        # "When the scalar_types argument is not [] things are more
+        # complicated. In most cases, using np.result_type and passing the
+        # Python values 0, 0.0, or 0j has the same result as using
+        # int, float, or complex in scalar_types."
+        types = []
         for arr, orig_arg in zip(arrs, orig_args):
             if arr.ndim == 0:
-                scalar_types.append(
-                    np.dtype(np.min_scalar_type(orig_arg))
-                    if use_min_scalar
-                    else arr.dtype
+                types.append(
+                    np.dtype(orig_arg) if use_min_scalar else arr.dtype
```

Comment:
```suggestion
                    orig_arg if use_min_scalar else arr.dtype
```
sorry, lets try this.  My thought was to keep the `.type(0)` on the first branch initially, but it will be wrong without NEP 50 finalized.  You definitely need the scalar though.
(This will call `np.asarray(orig_arg)` effectively, but the old code did too)

Using `orig_arg` for any 0-D input should mean you can skip the `use_min_scalar` logic (but that is the follow up once this works.)

Reply:
Still seeing failures, all see to be type mismatches